horovod/runner/elastic/discovery.py

```diff
@@ -169,7 +169,9 @@ def find_available_hosts_and_slots(self):
             if ':' in line:
                 host, slots = line.split(':')
                 host_slots[host] = int(slots)
-            else:
+            # Make sure the host is not empty. The discovery script might
+            # return empty string when all workers are not ready or available.
+            elif host:
```

Comment:
Can you add some tests? Also this would result in an empty map being returned to the caller (update_available_hosts) , which would result in the HostManager treating this as an update in the set of hosts, which is not intended.